### ISSUE TITLE
Android Layout : incorrect calculate width using flexWrap and flexBasis

### ISSUE THREAD

**Issue by Freddy03h** [facebook/react-native#11915]
### Description
Android has an issue with layout using `flexWrap` and `flexBasis`. It's inconsistent with the flexbox spec and the iOS result.

An item with a specified `flexBasis: X` and `flexShrink: 0` can't have a calculate width inferior to X. But on Android this unexpected behavior appear.

### Reproduction
```
  render() {

    return (
      <View
        style={{
          flexGrow: 1,
          backgroundColor: 'yellow',
          flexDirection: 'row',
          flexWrap: 'wrap',
          alignItems: 'flex-start',
        }}
        onLayout={(event) => console.log('container', event.nativeEvent.layout)}
      >
        {
          [0,1,2,3,4,5,6,7,8,9].map((i) =>
            <View
              key={i}
              style={{
                flexGrow: 1,
                flexShrink: 0,
                flexBasis: 130,
                backgroundColor: 'red',
                height: 195,
                borderWidth: 10,
                borderColor: 'blue',
              }}
              onLayout={(event) => console.log('item', event.nativeEvent.layout)}
            />
          )
        }
      </View>
    )
}
```

#### Android Marshmallow - Nexus 5X (device)
<img src="https://cloud.githubusercontent.com/assets/1412159/21962641/f5d40510-db2a-11e6-86d3-ea27080261cc.png" width="412" />

```
container {width: 411.4285583496094, height: 659.4285888671875}
item {width: 51.42856979370117, height: 195.04762268066406}
```

#### iOS 10.2 - iPhone SE (device)
<img src="https://cloud.githubusercontent.com/assets/1412159/21962632/ac37d698-db2a-11e6-8171-6e560ea33464.PNG" width="320" />

```
container {width: 320, height: 568}
item {width: 160, height: 195}
```

#### iOS 10.2 - iPhone 7 Plus (simulator)
<img src="https://cloud.githubusercontent.com/assets/1412159/21962695/7bcef9f8-db2c-11e6-8c87-f702c944710a.png" width="414" />

```
container {width: 414, height: 736}
item {width: 138, height: 195}
```

### Additional Information

* React Native version: 40
* Platform: Android 6.0

**Comment by Freddy03h** [comment]
Still got the issue with react-native `41.2`.

Should I open the issue on https://github.com/facebook/yoga ?

**Comment by Freddy03h** [comment]
> Yoga uses the same implementation on android and iOS so this is most likely not an issue with Yoga but with React Native.

Answer from Yoga's contributor.

**Comment by Freddy03h** [comment]
Fixed in RN 0.42.0 👌, you can close the issue.

Thank's for your amazing works ! 😃

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/android-layout--incorrect-calculate-width-using-flexwrap-and-flexbasis
